src/msw/webview_ie.cpp

```diff
@@ -853,25 +857,137 @@ wxString wxWebViewIE::GetPageText() const
+        return false;
+    }
+
+    if ( FAILED(document->get_Script(&scriptDispatch)) )
+    {
+        wxLogWarning(_("Can't get the Javascript"));
+        return false;
+    }
+
+    wxString counter;
+    wxString javaScriptVariable =
+        wxWebViewUtils::createVariableWithJavaScriptResult(javascript,
+        &m_runScriptCount, &counter);
+
+    wxAutomationObject scriptAO(scriptDispatch);
+    wxVariant varJavascript(javaScriptVariable);
+    wxVariant varResult;
+
+    if ( !scriptAO.Invoke("eval", DISPATCH_METHOD, varResult, 1, &varJavascript) )
+    {
+        wxLogWarning(_("Can't run Javascript"));
+        return false;
+    }
+
+    if ( varResult.IsType("bool") && varResult.GetBool() )
+    {
+        varJavascript = "(__wx$" + counter + " == null || \
+                        typeof __wx$" + counter + " != 'object') ? \
+                        String(__wx$" + counter + ") : __wx$" + counter;
+        if ( !scriptAO.Invoke("eval", DISPATCH_METHOD, varResult, 1, 
+            &varJavascript) )
         {
-            VARIANT level;
-            VariantInit(&level);
-            V_VT(&level) = VT_EMPTY;
-            window->execScript(wxBasicString(javascript),
-                               wxBasicString(language),
-                               &level);
+            wxLogWarning(_("Can't run Javascript"));
+            return false;
+        }
+
+        if ( varResult.IsType("void*") )
+        {   // Script returned an Object (JScriptTypeInfo IDispatch), convert it to JSON
+            IDispatch* dispatchResult = (IDispatch*)varResult.GetVoidPtr();
+            wxAutomationObject JSONAO;
+
+            if ( dispatchResult )
+            {
+                // JSON is not available in Quirks or IE6/7 standards mode,
+                // which is unfortunately the default one for the embedded
+                // browser control, see
+                // https://docs.microsoft.com/en-us/scripting/javascript/reference/json-object-javascript#requirements
+                // and see here how to make a program run use "modern" modes
+                // https://msdn.microsoft.com/en-us/library/ee330730(v=vs.85)#browser_emulation
+                if (scriptAO.GetObject(JSONAO, "JSON"))
+                {
+                    wxVariant varJSONStr;
+
+                    // Work around the bug in wxAutomationObject::Invoke(),
+                    // see https://trac.wxwidgets.org/ticket/14293
+                    dispatchResult->AddRef();
+                    if ( JSONAO.Invoke("stringify", DISPATCH_METHOD, varJSONStr, 1, &varResult) )
+                        varResult = varJSONStr;
+                    else
+		    {
+		        wxLogWarning(_("JSON.stringify fails when trying to convert object into JSON"));
+                        return false;
+		    }
+                }
+                else
+                {
+                    javaScriptVariable = " function __wx$stringifyJSON(obj) \
```

Comment:
Thanks for looking into the date issue, it seems it works now. 

I discovered another issue though. After running this script
```javascript
function Person(name) { this.name = name; }

function fn()
{
  Person.prototype.nationality = "English";
  return new Person("John");
}

fn();
```
your function returns  _{"name":"John","nationality":"English"}_ while IE's `JSON.stringify()` returns just _{"name":"John"}_. While intuitively your result seems to be the correct one, it appears that is not how JSON does it, see e.g. [here](https://stackoverflow.com/a/12369651).

IMHO (I am not a developer, just a random guy), I would rather avoid this can of worms which custom stringify is. I.e., if the programmer wants to get stringified objects, he must ensure that his webview app runs in a modern IE mode, otherwise only primitive values will be returned. But of course, this is for wxWidgets developers to decide...